test/unit/persist-queue.test.ts

```diff
@@ -9,7 +9,16 @@ import { walkSync } from '../e2e/utils';
 should();
 use(chaiAsPromised);
 
-function expectFilesInDirectory(dir: string, count: number): void {
+function delay(ms: number): Promise<void> {
+  return new Promise((resolve) => {
```

Comment:
Nit: should we preemptively put this in a `helpers.ts`?

Reply:
ah, almost certainly. I think we already have `delay` elsewhere.